rust/connlib/clients/shared/src/lib.rs

```diff
@@ -69,10 +77,14 @@ impl Session {
     ///
     /// In case of destructive network state changes, i.e. the user switched from wifi to cellular,
     /// reconnect allows connlib to re-establish connections faster because we don't have to wait for timeouts first.
-    pub fn reconnect(&mut self) {
+    pub fn reconnect(&self) {
         let _ = self.channel.try_send(Command::Reconnect);
     }
 
+    pub fn set_dns(&self, new_dns: Vec<IpAddr>) {
+        self.dns_updater.send(new_dns).expect("Developer error: As long as the session is up the dns update receiver must not be dropped");
```

Comment:
This can be hit if the app calls connlib after it emitted `on_disconnect`. I don't think we should panic here but just drop the message.

Reply:
You can make it a debug_assert if you want but there is no need to crash a production app I'd say.